Web/Edubase.Web.UI/Views/Tools/IndSchoolSearchDownload/ReadyToDownload.cshtml

```diff
@@ -53,7 +53,7 @@
                 <p>Your requested file is ready to download.</p>
 
                 <p>
-                    <input type="submit" class="govuk-button" data-module="govuk-button" value="@name" /><br />
+                    <input type="submit" id="@StringUtils.ElementIdFormat(@name)-button" class="govuk-button" data-module="govuk-button" value="@name" /><br />
```

Comment:
It would be worth double checking if this implementation is sufficient to meet this need, and would be robust enough. 

My thought process (and current assumption) is that `@name` refers to a file name or perhaps a display name?

If so, how does it handle apostrophes like for Children's Centres:

![image](https://user-images.githubusercontent.com/96429508/213682025-cd10ec51-6135-4f61-bd5d-ab4228b1fee0.png)

```csharp
// Current implementation:

namespace Edubase.Web.UI.Helpers
{
    public class StringUtils
    {
        public static string ElementIdFormat(string text)
        {
            return text.Replace(' ', '-').ToLowerInvariant();
        }
    }
}
```

```csharp
// Suggested implementation

using System;
using System.Text.RegularExpressions;
using Edubase.Web.UI.Helpers;


public class Program
{
    public static void Main()
    {
        Console.WriteLine("Hello World");
        Console.WriteLine(StringUtils.ElementIdFormat("example with spaces"));
        Console.WriteLine(StringUtils.ElementIdFormat("example with ' apostrophes"));
        Console.WriteLine(StringUtils.ElementIdFormat("example with \" quotes"));
        Console.WriteLine(StringUtils.ElementIdFormat("example with [] square brackets"));
        Console.WriteLine(StringUtils.ElementIdFormat("example with () round brackets"));
        Console.WriteLine(StringUtils.ElementIdFormat("example with accents -- The most common accents are the acute (é), grave (è), circumflex (â, î or ô), tilde (ñ), umlaut and dieresis (ü or ï – the same symbol is used for two different purposes), and cedilla (ç)."));
    }
}


namespace Edubase.Web.UI.Helpers
{
    public class StringUtils
    {
        public static string ElementIdFormat(string text)
        {
            var pattern = @"[^a-zA-Z0-9_]";
            var replacement = "-";
            var result = Regex.Replace(text, pattern, replacement);

            return result.ToLowerInvariant();
        }
    }
}
```


https://dotnetfiddle.net/X4b31o

![image](https://user-images.githubusercontent.com/96429508/213687162-77bd716b-0a1d-437f-80d6-90dd8cb55a75.png)

Reply:
Thanks @RogerHowellDfE this is great. I've made some updates to the StringUtils class in my latest PR but I will look to implement these changes as well.

Reply:
Ah, looks like @CathLass beat me to this comment (I was too busy tinkering 😁 ) and this has already been explained by @JaswiniP and resolved as a non-issue before I hit submit and refreshed the page -- I withdraw my comment!

For further info/context, it seems that `name` refers to the file type name from a constrained list of values, which should never have unusual characters so this PR is fine for this specific page 👍 

![image](https://user-images.githubusercontent.com/96429508/213690703-19ab738c-312b-48c2-85bd-9525c83c3d49.png)